apps/solution/src/contexts/PeopleContext.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useReducer } from "react";
+import { useContext } from "react";
+import { createContext } from "react";
+import { getPeople, PersonModel } from "../api/person";
+
+
+interface PeopleContextTypes {
+  people: PersonModel[],
+  dispatch: React.Dispatch<Action>
+}
+
+export const PeopleContext = createContext<PeopleContextTypes>({} as PeopleContextTypes);
+
+interface childrenProps {
+  children: JSX.Element[] | JSX.Element;
+}
+
+type PeopleState = PersonModel[]
+export enum PeopleActionKind {
+  ADD = 'ADD',
+  UPDATE = 'UPDATE',
+  REMOVE = 'REMOVE',
+  INIT = 'INIT'
+}
+interface PeopleAction { type: PeopleActionKind.ADD | PeopleActionKind.INIT, people: PersonModel[]; }
+interface PersonAction { type: PeopleActionKind.UPDATE | PeopleActionKind.REMOVE, person: PersonModel; }
+type Action = PeopleAction | PersonAction
```

Comment:
ça me parait un peu difficile à lire. Est-ce que ça ne serait pas plus facile à lire de lister chaque action séparément ?

```suggestion
type Action = 
  { type: 'INIT' as const, people: PersonModel[] }
  | { type: 'ADD' as const, people: PersonModel[] }
  | { type: 'UPDATE' as const, person: PersonModel }
  | { type: 'DELETE' as const, person: PersonModel}
```

Reply:
Mmh dans ce cas les typage d'action ne servent plus trop non ? J'ai aussi cru comprendre qu'il faut éviter les "as" ?